### DISCUSSION TITLE
Push button is highlighted with purple color instead of pull button, if there are commits available to pull.

### DISCUSSION THREAD

**Discussion by NikolayGrigoryev** [gitextensions/gitextensions#10809] [Q&A] [answered]
### Environment

- Git Extensions 4.0.2.16100
- Build 25100ec1f7c8da8f798613da74826038af81a50e
- Git 2.33.0.windows.2 (recommended: 2.38.1 or later)
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.13
- DPI 96dpi (no scaling)
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 5.0.14 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Issue description

When in the current branch are available more commits to pull, push button is highlighted instead of pull button.
![image](https://user-images.githubusercontent.com/86227400/225877061-ec4fb4e3-2d44-468a-bc61-eeeb90ca8151.png)

### Steps to reproduce

Check out to existing local branch with more commits available in the remote branch.

### Did this work in previous version of GitExtensions?

I think it exists from 4.0

**Comment by gerhardol** [comment]
GE can only know about commits and remote branches for what have been fetched,
The purple button and the info on the button show that the local branch is one commit behind, so if you push you will reset the remote branch,

Putting this info on pull would be incorrect, as you do not know if any commits have been added at the remote.

**Comment by pmiossec** [comment]
See  #5994 for more details...

The color is there to warn/inform the user that he normally has to do an action (merge or rebase) before he will be able to push because he is at the same time ahead and behind or he will have to do a force push if that is really what he wants to do. See tooltip for more information.

**Comment by NikolayGrigoryev** [comment]
In this case I can suggest a change to UX for more warning look like:
![image](https://user-images.githubusercontent.com/86227400/225889956-42f5187a-88b3-4647-bbb3-6ea37482b13a.png)

**Comment by NikolayGrigoryev** [comment]
I can guess usually force pushes are restricted by default. For less experienced user it is not always clear that push is still available, but it is safer to do pull before and resolve possible collisions. For example, looking to earlier SourceTree versions, people used to do pull if they see commits to pull, before push. In the subject, purple color on push is much more noticeable than arrow with number at right.

**Comment by RussKie** [comment, marked as answer]
Personally, I've never paid attention to the color of the button - i push
and let the app tell me when I'm rewriting history. However,  I'm not sure
having an exclamation mark would facilitate the right metaphor, to me that
looks like something is wrong and I may not be able to push...

Git will reject a push that would result in lost commits, and I'll have to
be explicit about force pushing.
